src/main/scala/com/ambiata/disorder/GenPlus.scala

```diff
@@ -18,4 +18,26 @@ object GenPlus {
    */
   def listOfSized[A](from: Int, to: Int, gen: => Gen[A]): Gen[List[A]] =
     Gen.sized(n => Gen.choose(from, Math.max(from, Math.min(n, to))).flatMap(i => Gen.listOfN(i, gen)))
+
+  /*
+   Safely generate an arbitrary which matches a predicate. This function will recursively
+   attempt to create the arbitrary until it matches the predicate without failing, unless
+   the number of attempts is greater than 100.
+   */
+  def checkGen[A](n: Int, gen: Gen[A], label: String, check: A => Boolean): Gen[A] = {
+    if (n > 100) Gen.fail.label(s"Couldn't generate a $label")
+    else for {
+      a <- gen
+      z <- Gen.size
+      r <- if (check(a)) Gen.const(a) else Gen.resize(z + 1, checkGen[A](n + 1, gen, label, check))
+    } yield r
+  }
+
+  def checkGenTuple[A, B: Arbitrary, C: Arbitrary](make: (B, C) => A, label: String, check: A => Boolean): Gen[A] = for {
```

Comment:
Saves a bunch of duplicate client code in DistinctPair/EncodingString for example